### ISSUE TITLE
Loss values often go to nan and break the trainer

### ISSUE THREAD

**Issue by schwab** [werner-duvaud/muzero-general#118]
```
last test reward: 1622.00. Training step: 4/10800. Played games: 1. Loss: nan.08

```

Once this happens the training is on longer working.  Sometimes adjusting the lr_init can help, but often it breaks again when the number of CNN Blocks or channels is changed.

```
after a while this is dumped....

```
(pid=3692069) Moves: 116, ticks: 807 ratio: 0.1400. Played games: 1. Loss: nan08
(pid=3692077) /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [1,0,0], thread: [32,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
(pid=3692077) /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [1,0,0], thread: [33,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
(pid=3692077) /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [1,0,0], thread: [34,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
(pid=3692077) /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [1,0,0], thread: [35,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
(pid=3692077) /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [1,0,0], thread: [36,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
(pid=3692077) /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [1,0,0], thread: [37,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
(pid=3692077) /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [1,0,0], thread: [38,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
(pid=3692077) /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [0,0,0], thread: [17,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
(pid=3692077) /pytorch/aten/src/ATen/native/cuda/ScatterGatherKernel.cu:115: operator(): block: [0,0,0], thread: [18,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
```

**Comment by ahainaut** [comment]
Hi @schwab ,
Sorry for the late response. Could you please provide more informations about your configuration so I can try to reproduce your error ?

**Comment by schwab** [comment]
Looks like Jason-ZhouZJ  reproduced the same problem in issue #123.  Mine is a different custom game that my son wrote, but the behavior was similar.  I finally surmised that issue was due to the learning rate being too large and the fact that it takes a very long time to play enough games.   The setup we have only allows us to play ~ 2 games at a time. By reducing the LR by several orders of magnitude I was able to eliminate the NaN's.  Nonetheless even after a full week of training the algorithm is barely able to beat even a random choosing opponent.  When interacting with a human it's barely better than a rage doll.

**Comment by ahainaut** [comment]
Glad you could solve the Nan issue ! If you need help with training Muzero on a custom environment you can go on the [discord page](https://discord.gg/GB2vwsF) to ask other members, maybe someone had already trained Muzero on a similar environment !
As you have solved your problem, I'll close this issue.
